docs/architecture/adr-047-extend-upgrade-plan.md

```diff
@@ -133,76 +100,68 @@ For example, if the `url` is `"https://example.com?checksum=md5:d41d8cd98f00b204
+    3. The `url` field MUST be a proper URL.
+    4. A `checksum` MUST be provided either in the `checksum` field or as a query parameter in the `url`.
+    5. If the `checksum` field has a value and the `url` also has a `checksum` query parameter, the two values MUST be equal.
+    6. The `url` MUST return either a file or an archive containing either `bin/{DAEMON_NAME}` or `{DAEMON_NAME}`.
+    7. If a `checksum` is provided (in the field or as a query param), the checksum of the result of the `url` MUST equal the provided checksum.
 
 Downloading of an `Artifact` will happen the same way that URLs from `info` are currently downloaded.
 
 ### Cosmovisor Updates
 
-If the `upgrade-info.json` file does not contain any `UpgradeInstructions`, existing functionality will be maintained.
+If the `upgrade-info.json` file does not contain any `Artifacts`, existing functionality is maintained.
 
-We will update Cosmovisor to look for and handle the new `UpgradeInstructions` in `upgrade-info.json`.
-If the `UpgradeInstructions` are provided, we will do the following:
+Cosmovisor is to be updated to look for and handle the new `Artifacts` in `upgrade-info.json`.
+If the `Artifacts` are provided, the following should be done:
 
-1.  The `info` field will be ignored.
-1.  The `artifacts` field will be used to identify the artifact to download based on the `platform` that Cosmovisor is running in.
-1.  If a `checksum` is provided (either in the field or as a query param in the `url`), and the downloaded artifact has a different checksum, the upgrade process will be interrupted and Cosmovisor will exit with an error.
-1.  If a `pre_run` command is defined, it will be executed at the same point in the process where the `app pre-upgrade` command would have been executed.
-    It will be executed using the same environment as other commands run by Cosmovisor.
-1.  If a `post_run` command is defined, it will be executed after executing the command that restarts the chain.
-    It will be executed in a background process using the same environment as the other commands.
-    Any output generated by the command will be logged.
-    Once complete, the exit code will be logged.
+1. The `info` field is ignored.
+2. The `artifacts` field is used to identify the artifact to download based on the `platform` that Cosmovisor is running in.
+3. If a `checksum` is provided (either in the field or as a query param in the `url`), and the downloaded artifact has a different checksum, the upgrade process is interrupted and Cosmovisor will exit with an error.
 
-We will deprecate the use of the `info` field for anything other than human readable information.
-A warning will be logged if the `info` field is used to define the assets (either by URL or JSON).
+The `info` field should be deprecated for anything other than human readable information.
+A warning is logged if the `info` field is used to define the assets (either by URL or JSON).
```

Comment:
After thinking about this for the last year+, I don't think this deprecation is a good idea. The specific functionality being removed that is still needed, is the ability to provide a URL that returns the artifact download URLs. The new `artifacts` stuff requires the artifact download URLs to be directly in the upgrade plan.